include/mp/proxy-types.h

```diff
@@ -389,30 +389,17 @@ struct ClientParam
 
     struct BuildParams : IterateFieldsHelper<BuildParams, sizeof...(Types)>
     {
-        template <typename... Args>
-        void handleField(Args&&... args)
-        {
-            callBuild<0>(std::forward<Args>(args)...);
-        }
-
-        // TODO Possible optimization to speed up compile time:
-        // https://stackoverflow.com/a/7858971 Using enable_if below to check
-        // position when unpacking tuple might be slower than pattern matching
-        // approach in the stack overflow solution
-        template <size_t I, typename... Args>
-        auto callBuild(Args&&... args) -> std::enable_if_t<(I < sizeof...(Types))>
-        {
-            callBuild<I + 1>(std::forward<Args>(args)..., std::get<I>(m_client_param->m_values));
-        }
-
-        template <size_t I, typename Params, typename ParamList, typename... Values>
-        auto callBuild(ClientInvokeContext& invoke_context, Params& params, ParamList, Values&&... values) ->
-            std::enable_if_t<(I == sizeof...(Types))>
+        template <typename Params, typename ParamList>
+        void handleField(ClientInvokeContext& invoke_context, Params& params, ParamList)
         {
-            MaybeBuildField(std::integral_constant<bool, Accessor::in>(), ParamList(), invoke_context,
-                Make<StructField, Accessor>(params), std::forward<Values>(values)...);
-            MaybeSetWant(
-                ParamList(), Priority<1>(), std::forward<Values>(values)..., Make<StructField, Accessor>(params));
+            auto const fun = [&]<typename... Values>(Values&&... values) {
```

Comment:
In commit "replace custom tuple unpacking code with `std::apply`" (36898dbfc72c9748f54bf79f18d1ad6485f48920)

I think this code might be more straightforward and easier to review if this got rid of the `fun` variables and just passed the lambda directly as the first argument to std::apply. That way the indentation of the code inside the lambda would not have to change and the diff would be smaller and more obvious, and the resulting code would also be shorter.

Reply:
> That way the indentation of the code inside the lambda would not have to change

It does. Using the lambda directly has no effect on the number of `{` and hence the indentation.
Also, running `git clang-format` after trying that completely borked the formatting due to no `.clang-format`.

I prefer passing lambda expression to functions directly **iff** the expression is small and it is the last function argument. Here, the lambda expression is complex and it is not the last function argument. Giving it a name results in longer, but more readable code.